_studio/mfx_lib/shared/src/mfx_h264_encode_vaapi.cpp

```diff
@@ -251,6 +249,29 @@ mfxStatus SetRateControl(
     return MFX_ERR_NONE;
 }
 
+mfxStatus SetBrcResetRequired(
+    VADisplay             vaDisplay,
+    VABufferID &          rateParamBuf_id,
+    bool                  isBrcResetRequired)
+{
+    VAStatus vaSts;
+    VAEncMiscParameterBuffer *misc_param;
+    VAEncMiscParameterRateControl *rate_param;
+
+    vaSts = vaMapBuffer(vaDisplay,
```

Comment:
Need to check if rateParamBuf_id was created ?
Like:
mfxStatus SetBrcResetRequired(
...
if (rateParamBuf_id != VA_INVALID_ID)
{
    vaMapBuffer()
    rate_param->rc_flags.bits.reset = isBrcResetRequired;
    vaUnmap()
}
}

Reply:
Good catch. It should be something:
```
MFX_CHECK(rateParamBuf_id != VA_INVALID_ID, MFX_ERR_INVALID_HANDLE);
```